pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -424,3 +431,26 @@ func (p *Provider) DeleteLaunchTemplates(ctx context.Context, nodeClass *v1beta1
 	}
 	return nil
 }
+
+func (p *Provider) ResolveClusterCIDR(ctx context.Context) error {
+	if p.ClusterCIDR != nil {
+		return nil
+	}
+	out, err := p.eksapi.DescribeClusterWithContext(ctx, &eks.DescribeClusterInput{
+		Name: aws.String(options.FromContext(ctx).ClusterName),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to resolve cluster CIDR, %w", err)
+	}
+	if ipv4CIDR := out.Cluster.KubernetesNetworkConfig.ServiceIpv4Cidr; ipv4CIDR != nil {
+		p.ClusterCIDR = ipv4CIDR
+		logging.FromContext(ctx).Debugf("discovered cluster CIDR %q", p.ClusterCIDR)
+		return nil
+	}
+	if ipv6CIDR := out.Cluster.KubernetesNetworkConfig.ServiceIpv6Cidr; ipv6CIDR != nil {
+		logging.FromContext(ctx).Debugf("discovered cluster CIDR %q", p.ClusterCIDR)
+		p.ClusterCIDR = ipv6CIDR
+		return nil
+	}
+	return fmt.Errorf("failed to resolve cluster CIDR")
```

Comment:
```suggestion
	return fmt.Errorf("resolving cluster CIDR, no CIDR found in DescribeCluster response")
```

nit: Be more specific with the error, also consider using the gerund here